spec.html

```diff
@@ -303,17 +310,39 @@ <h1>SharedArrayBuffer.prototype.grow ( _newLength_ )</h1>
         1. Let _O_ be the *this* value.
         1. Perform ? RequireInternalSlot(_O_, [[ArrayBufferMaxByteLength]]).
         1. If IsSharedArrayBuffer(_O_) is *false*, throw a *TypeError* exception.
-        1. Let _newByteLength_ to ? ToIntegerOrInfinity(_newLength_).
-        1. If _newByteLength_ &lt; _currentByteLength_ or _newByteLength_ &gt; _O_.[[ArrayBufferMaxByteLength]], throw a *RangeError* exception.
         1. Let _hostHandled_ be ? HostGrowSharedArrayBuffer(_O_, _newByteLength_).
         1. If _hostHandled_ is ~handled~, return *undefined*.
-        1. Let _currentByteLength_ be ArrayBufferByteLength(_O_, ~SeqCst~).
-        1. If _newByteLength_ &ne; _currentByteLength_, then
-          1. Note: Resizes to the same length explicitly do nothing to avoid gratuitous synchronization.
-          1. Let _byteLengthBlock_ be _O_.[[ArrayBufferByteLengthData]].
-          1. Perform SetValueInBuffer(_byteLengthBlock_, 0, ~Uint64~, 𝔽(_newByteLength_), *true*, ~SeqCst~).
+        1. Let _newByteLength_ to ? ToIntegerOrInfinity(_newLength_).
+        1. Let _rawCurrentByteLengthBytesRead_ be a List of length 8 whose elements are nondeterministically chosen byte values.
+        1. NOTE: In implementations, _rawCurrentByteLengthBytesRead_ is the result of a load-link, of a load-exclusive, or of an operand of a read-modify-write instruction on the underlying hardware. The nondeterminism is a semantic prescription of the memory model to describe observable behaviour of hardware with weak consistency.
+        1. Let _byteLengthBlock_ be _O_.[[ArrayBufferByteLengthData]].
+        1. Let _isLittleEndian_ be the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+        1. Let _currentBytesLength_ be RawBytesToNumeric(~BigUint64~, _rawCurrentByteLengthBytesRead_, _isLittleEndian_).
```

Comment:
_currentByteLength_ instead of _currentBytesLength_

Reply:
Good eye, thanks for spotting.